connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala

```diff
@@ -1917,6 +1991,37 @@ class SparkConnectPlanner(val session: SparkSession) {
     }
   }
 
+  private def transformTypedReduceExpression(
+      fun: proto.Expression.UnresolvedFunction,
+      dataAttributes: Seq[Attribute]): Expression = {
+    assert(fun.getFunctionName == "reduce")
+    if (fun.getArgumentsCount != 1) {
+      throw InvalidPlanInput("reduce requires single child expression")
+    }
+    val udf = fun.getArgumentsList.asScala.toSeq.head match {
+      case expr
+          if expr.hasCommonInlineUserDefinedFunction
+            && expr.getCommonInlineUserDefinedFunction.hasScalarScalaUdf =>
+        TypedScalaUdf(expr.getCommonInlineUserDefinedFunction)
+      case other =>
+        throw InvalidPlanInput(s"reduce should be a scalar scala udf, but got $other")
```

Comment:
Does it really have to be a typed scala udf? I mean it just needs to a 2-ary UDF.